src/python/pants/backend/javascript/subsystems/nodejs.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Iterable
+
+from pants.core.util_rules.external_tool import (
+    DownloadedExternalTool,
+    ExternalToolRequest,
+    TemplatedExternalTool,
+)
+from pants.engine.fs import EMPTY_DIGEST, Digest, MergeDigests
+from pants.engine.platform import Platform
+from pants.engine.process import Process
+from pants.engine.rules import Get, Rule, SubsystemRule, collect_rules, rule
+from pants.engine.unions import UnionRule
+from pants.util.logging import LogLevel
+
+
+class NodeJS(TemplatedExternalTool):
+    options_scope = "nodejs"
+    help = "The NodeJS Javascript runtime (including npm and npx)."
+
+    # TODO: Update to latest LTS
+    default_version = "v16.14.1"
+    default_known_versions = [
+        "v16.14.1|macos_arm64|8f6d45796f3d996484dcf53bb0e53cd019cd0ef7a1a247bd0178ebaa7e63a184|28983951",
+        "v16.14.1|macos_x86_64|af35abd727b051c8cdb8dcda9815ae93f96ef2c224d71f4ec52034a2ab5d8b61|30414480",
+        "v16.14.1|linux_arm64|53aeda7118cd1991424b457907790033326f432ee6c2908a7693920124622cf4|32852829",
+        "v16.14.1|linux_x86_64|8db3d6d8ecfc2af932320fb12449de2b5b76f946ac72b47c6a9074afe82737ff|32861678",
+    ]
+
+    default_url_template = "https://nodejs.org/dist/{version}/node-{version}-{platform}.tar.gz"
+    default_url_platform_mapping = {
+        "macos_arm64": "darwin-arm64",
+        "macos_x86_64": "darwin-x64",
+        "linux_arm64": "linux-arm64",
+        "linux_x86_64": "linux-x64",
+    }
+
+    def generate_exe(self, plat: Platform) -> str:
+        assert self.default_url_platform_mapping is not None
+        plat_str = self.default_url_platform_mapping[plat.value]
+        return f"./node-{self.version}-{plat_str}/bin/node"
+
+
+@dataclass(frozen=True)
+class NpxProcess:
```

Comment:
I originally had a "base NPX Tool" concept, but that looked and felt wrong - https://github.com/sureshjoshi/pants-plugins/blob/00208b3a9c7dff2e53d47e25c626bbb7b8a42b59/pants-plugins/experimental/nodejs/subsystem.py#L38-L54

So, I switched to an NpxProcess, which feels better in usage - since all we're doing is calling out to NPX via Node to run the subsequent arguments + package.

Bikeshed: "npm_package" - I don't like that name, since we're already in the NPX class-space, I feel that already implies npm. I was thinking maybe "package", "package_name", or something along those lines.

Bikeshed: "argv" or "args"?

Reply:
I like NPXProcess - lean into the abstraction. It is more than an "implementation detail" to hide, it's foundational.

> npm_package

Seems fine to me because it implies the package must be available on NPM.

> Bikeshed: "argv" or "args"?

Usually I'm +1 for `argv`, but probably `args` because this is not a true `argv`. We insert a couple args before for you. Wdyt?

Reply:
I originally wrote `argv` because it included the process command, then I re-wrote a bunch of things and changed it to `args`, then I got confused last night and went back to `argv` for some reason (I think to align with PexProcess maybe). So, I punted it over to a question here :)

I'll change it back to args, as that feels more correct in this case with it coming after a node + npx + npm_package command.